src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -973,7 +978,12 @@ protected internal virtual unsafe void Reset(bool forceReleaseLongLivedAllocator
             for (var i = 0; i < _numberOfAllocatedStringsValues; i++)
                 _allocateStringValues[i].Reset();
 
-            _perCoreLazyStringValuesList.Push(_allocateStringValues);
+            if (_perCoreLazyStringValuesList.TryPush(_allocateStringValues) == false)
```

Comment:
Need to check that this isn't already null. 
Pretty sure that this is why we are getting test failures.